test-suite/testheaders.sh

```diff
@@ -1,51 +0,0 @@
-#!/bin/sh
-#
-## Copyright (C) 1996-2023 The Squid Software Foundation and contributors
-##
-## Squid software is distributed under GPLv2+ license and includes
-## contributions from numerous individuals and organizations.
-## Please see the COPYING and CONTRIBUTORS files for details.
-##
-
-# test all header files (.h) for dependency issues.
-#
-# Ideally this test should be performed twice before any code is accepted.
-# With or without inline enabled.  This is needed because the .cci files
-#  are only included into the .h files when inline mode is enabled.
-#
-# This script should be run from the makefile with the directory path and ccflags
-#
-cc="${1}"
-shift
-for dir in /usr/bin /usr/local/bin /usr/gnu/bin
-do
-	test -x ${dir}/true && TRUE=${dir}/true
-done
-TRUE=${TRUE:-/bin/true}
-
-exitCode=0
-
-for f in $@; do
-	echo -n "Testing ${f} ..."
-    t="testhdr_`basename ${f}`"
-    if [ ! -f "$t.o" -o $f -nt "$t.o" ]; then
-        echo >$t.cc <<EOF
```

Comment:
I have added the following text to the PR description: Since 2010 commit https://github.com/squid-cache/squid/commit/a0fdc9bf8ff41a49ac384d90f2c33aef62db3abf, existing testHeaders
were not detecting problems in any headers due to a use of a shell heredoc with echo, a command that does not read from standard input.

I have not found other `echo <<HEREDOC` bugs in Squid sources.

Reply:
Thanks for the debugging exercise! What a pity for the findings :(